tests/ui/job-view/fuzzy_test.jsx

```diff
@@ -0,0 +1,235 @@
+    title: 'Syrup',
+    author: {
+      firstName: 'Max',
+      lastName: 'Barry',
+    },
+  },
+  {
+    title: 'The Lost Symbol',
+    author: {
+      firstName: 'Dan',
+      lastName: 'Brown',
+    },
+  },
+  {
+    title: 'The Book of Lies',
+    author: {
+      firstName: 'Brad',
+      lastName: 'Meltzer',
+    },
+  },
+  {
+    title: 'Lamb',
+    author: {
+      firstName: 'Christopher',
+      lastName: 'Moore',
+    },
+  },
+  {
+    title: 'Fool',
+    author: {
+      firstName: 'Christopher',
+      lastName: 'Moore',
+    },
+  },
+  {
+    title: 'Incompetence',
+    author: {
+      firstName: 'Rob',
+      lastName: 'Grant',
+    },
+  },
+  {
+    title: 'Fat',
+    author: {
+      firstName: 'Rob',
+      lastName: 'Grant',
+    },
+  },
+  {
+    title: 'Colony',
+    author: {
+      firstName: 'Rob',
+      lastName: 'Grant',
+    },
+  },
+  {
+    title: 'Backwards, Red Dwarf',
+    author: {
+      firstName: 'Rob',
+      lastName: 'Grant',
+    },
+  },
+  {
+    title: 'The Grand Design',
+    author: {
+      firstName: 'Stephen',
+      lastName: 'Hawking',
+    },
+  },
+  {
+    title: 'The Book of Samson',
+    author: {
+      firstName: 'David',
+      lastName: 'Maine',
+    },
+  },
+  {
+    title: 'The Preservationist',
+    author: {
+      firstName: 'David',
+      lastName: 'Maine',
+    },
+  },
+  {
+    title: 'Fallen',
+    author: {
+      firstName: 'David',
+      lastName: 'Maine',
+    },
+  },
+  {
+    title: 'Monster 1959',
+    author: {
+      firstName: 'David',
+      lastName: 'Maine',
+    },
+  },
+];
+const searchDavid = [
+  {
+    title: 'The Book of Samson',
+    author: {
+      firstName: 'David',
+      lastName: 'Maine',
+    },
+  },
+  {
+    title: 'The Preservationist',
+    author: {
+      firstName: 'David',
+      lastName: 'Maine',
+    },
+  },
+  {
+    title: 'Fallen',
+    author: {
+      firstName: 'David',
+      lastName: 'Maine',
+    },
+  },
+  {
+    title: 'Monster 1959',
+    author: {
+      firstName: 'David',
+      lastName: 'Maine',
+    },
+  },
+  {
+    title: 'The DaVinci Code',
+    author: {
+      firstName: 'Dan',
+      lastName: 'Brown',
+    },
+  },
+  {
+    title: 'Angels & Demons',
+    author: {
+      firstName: 'Dan',
+      lastName: 'Brown',
+    },
+  },
+  {
+    title: 'The Lost Symbol',
+    author: {
+      firstName: 'Dan',
+      lastName: 'Brown',
+    },
+  },
+  {
+    title: 'The Grand Design',
+    author: {
+      firstName: 'Stephen',
+      lastName: 'Hawking',
+    },
+  },
+];
+test('Fuzzy search gives expected results', async () => {
+  const options = {
+    // http://fusejs.io/ describes the options available
+    keys: ['title', 'author.firstName'],
+    threshold: 0.6, // This seems like a good threshold to remove most false matches, lower is stricter
+    matchAllTokens: true,
+    tokenize: true,
+  };
+  // expected results on searching David
+  const fuse = new Fuse(testFuseArray, options);
```

Comment:
I'm curious why we need this test. :) It looks like you're just testing that a function from an external library works, rather than testing a component in our code base that uses it?

Reply:
We want to upgrade fuse.js and wanted to have tests for .I also had an initial thought similar to as you said @sarah-clements 
I think @armenzg can give the appropriate answer

Reply:
If you're comfortable to rendering a component and using react testing library to do the testing feel free to go for it.

I was not sure how comfortable you're jumping at the deep end.

Reply:
If it helps, [this](https://gist.githubusercontent.com/KWierso/fdcf8fb2013da29fb5515ce47b0acaf8/raw/fbe8458ef2d0bf4f93ebd6fcac45a3756ff8aad6/filteredFuzzyList.json) is the list of jobs (and the format of that list of jobs) that Treeherder's `Add New Jobs (Search)` dialog uses with Fuse to search for jobs. 

That file is massive (2+MB), with 10,000+ jobs included. You could probably make a much smaller version of it and still get the testing you want from this.

Reply:
Thanks a lot @KWierso 
Shall I use upgrade fuze.js in this too and use options for fuzzy search from https://github.com/mozilla/treeherder/pull/6125
which are

1. change threshhold from 0.6 to 0.4
2. use extended search:true
@armenzg